tests/bsim/bluetooth/host/adv/extended/src/ext_adv_advertiser.c

```diff
@@ -168,12 +169,23 @@ static struct bt_conn_cb conn_cbs = {
 static void main_ext_adv_advertiser(void)
 {
 	struct bt_le_ext_adv *ext_adv;
+	uint8_t adv_handle;
+	int err;
 
 	common_init();
 
 	create_ext_adv_set(&ext_adv, false);
 	start_ext_adv_set(ext_adv);
 
+	err = bt_hci_get_adv_handle(ext_adv, &adv_handle);
+	if (err) {
+		FAIL("Failed to obtain adv handle: %d\n", err);
+	}
+
+	if (bt_adv_lookup_handle(adv_handle) != ext_adv) {
+		FAIL("Adv handle lookup failed");
+	}
```

Comment:
I think these are better suited for unittests, rather than BSIM tests

Reply:
Yes, you are right. Do you have a sample unit tests that can be used as a starting point?

Reply:
I think any of the tests in https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/bluetooth/host are suitable for that